org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageClientImpl.java

```diff
@@ -83,6 +95,54 @@ public final void logMessage(MessageParams message) {
 		CompletableFuture.runAsync(() -> ServerMessageHandler.logMessage(wrapper, message));
 	}
 
+	private Map<String, BlockingQueue<ProgressParams>> progressQueueMap = new HashMap<>();
+
+	@Override
+	public CompletableFuture<Void> createProgress(WorkDoneProgressCreateParams params) {
+		String jobName = params.getToken().getLeft();
+		LinkedBlockingDeque<ProgressParams> blockingDeque = new LinkedBlockingDeque<ProgressParams>();
+		progressQueueMap.put(jobName, blockingDeque);
+
+		Job job = Job.create(jobName, new ICoreRunnable() {
+			@Override
+			public void run(IProgressMonitor monitor) throws CoreException {
+				while (true) {
```

Comment:
I'm a bit uncomfortable with `while (true)` what about `while((nextNotification = blockingQueue.take()) != null)` ?

Reply:
Thanks, that is much better.
I will continue working on it.